api/tests/core/users/email/test_update.py

```diff
@@ -219,3 +221,133 @@ def test_email_update_cancellation_suspend_account_failed(self, app):
 
         # Token is not deleted
         assert app.redis_client.get(email_update.get_token_key(user, email_update.TokenType.CONFIRMATION)) is not None
+
+
+class EmailUpdateTokenTest:
+    @pytest.mark.parametrize(
+        "token_type",
+        [
+            (email_update.TokenType.CONFIRMATION),
+            (email_update.TokenType.VALIDATION),
+        ],
+    )
+    def test_generate_email_change_token(self, app, token_type):
+        # Given
+        user = users_factories.UserFactory()
+        new_email = "Updated_" + user.email
+        expiration_date = email_update.generate_email_change_token_expiration_date()
+
+        # When
+        token = email_update.generate_email_change_token(
+            user, new_email, expiration_date=expiration_date, token_type=token_type
+        )
+
+        # Then
+        decoded_token = decode_jwt_token(token)
+        assert app.redis_client.get(email_update.get_token_key(user, token_type)) == token
+        assert decoded_token["current_email"] == user.email
+        assert decoded_token["new_email"] == new_email
+        assert decoded_token["exp"] == int(expiration_date.timestamp())
+
+    @pytest.mark.parametrize(
+        "token_type",
+        [
+            (email_update.TokenType.CONFIRMATION),
+            (email_update.TokenType.VALIDATION),
+        ],
+    )
+    def test_check_token_nominal(self, app, token_type):
+        # Given
+        user = users_factories.UserFactory()
+        new_email = "Updated_" + user.email
+        expiration_date = email_update.generate_email_change_token_expiration_date()
+        token = email_update.generate_email_change_token(
+            user, new_email, expiration_date=expiration_date, token_type=token_type
+        )
+
+        # When
+        email_update._check_token(user, token, token_type)
+
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** Tu as perdu des `assert` ici je pense

Reply:
je ne sais pas ce que je doit assert. check token ne doit pas causer d'erreur c'est l'objectif.

Reply:
Ok je vois
Dans ce cas le commentaire `#When` est misleading
tu peux mettre `should not raise`